File: update_theme.py
import os
import re

directory = r"d:\my projects\electrical\sres-website\src\app"

replacements = [
    (r'blue-950', 'brand-dark'),
    (r'blue-900', 'brand-dark'),
    (r'blue-800', 'brand-dark'),
    (r'blue-700', 'brand-dark'),
    (r'blue-600', 'brand-primary'),
    (r'blue-500', 'brand-primary'),
    (r'blue-400', 'brand-primary'),
    (r'blue-300', 'brand-secondary'),
    (r'blue-200', 'brand-primary/40'),
    (r'blue-100', 'brand-primary/20'),
    (r'blue-50', 'brand-primary/10'),
    (r'indigo-700', 'brand-dark'),
    (r'indigo-600', 'brand-secondary'),
    (r'indigo-500', 'brand-secondary'),
    (r'indigo-400', 'brand-secondary'),
    (r'from-blue', 'from-brand-primary'),
    (r'via-blue', 'via-brand-primary'),
    (r'to-blue', 'to-brand-primary'),
    (r'from-indigo', 'from-brand-secondary'),
    (r'to-indigo', 'to-brand-secondary'),
]

for root, dirs, files in os.walk(directory):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            new_content = content
            for old_patt, new_patt in replacements:
                # To prevent replacing standard text, we can just do simple string replacements
                # for tailwind classes like 'bg-blue-600' -> 'bg-brand-primary'
                # but 'blue-600' is unique enough
                new_content = re.sub(r'(?<!\w)' + old_patt + r'(?!\w)', new_patt, new_content)
                new_content = new_content.replace(old_patt + '/', new_patt + '/')
            
            # Additional targeted fix for layout.tsx's selection text
            new_content = new_content.replace('selection:bg-brand-primary', 'selection:bg-brand-primary selection:text-black')
            
            if content != new_content:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"Updated {filepath}")

File: update_theme.js
const fs = require('fs');
const path = require('path');

const directory = path.join(__dirname, 'src', 'app');

const replacements = [
    { old: /blue-950/g, new: 'brand-dark' },
    { old: /blue-900/g, new: 'brand-dark' },
    { old: /blue-800/g, new: 'brand-dark' },
    { old: /blue-700/g, new: 'brand-dark' },
    { old: /blue-600/g, new: 'brand-primary' },
    { old: /blue-500/g, new: 'brand-primary' },
    { old: /blue-400/g, new: 'brand-primary' },
    { old: /blue-300/g, new: 'brand-secondary' },
    { old: /blue-200/g, new: 'brand-primary/40' },
    { old: /blue-100/g, new: 'brand-bg/50' },
    { old: /blue-50/g, new: 'brand-bg/20' },
    { old: /indigo-700/g, new: 'brand-dark' },
    { old: /indigo-600/g, new: 'brand-secondary' },
    { old: /indigo-500/g, new: 'brand-secondary' },
    { old: /indigo-400/g, new: 'brand-secondary' },
    { old: /from-blue/g, new: 'from-brand-primary' },
    { old: /via-blue/g, new: 'via-brand-primary' },
    { old: /to-blue/g, new: 'to-brand-primary' },
    { old: /from-indigo/g, new: 'from-brand-secondary' },
    { old: /to-indigo/g, new: 'to-brand-secondary' },
    { old: /selection:bg-blue-600\b/g, new: 'selection:bg-brand-primary selection:text-black' },
];

function processDirectory(dir) {
    const files = fs.readdirSync(dir);
    
    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        
        if (stat.isDirectory()) {
            processDirectory(fullPath);
        } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            let newContent = content;
            
            for (const { old, new: replacement } of replacements) {
                newContent = newContent.replace(old, replacement);
            }
            
            // Fix any weird outputs from chained replaces
            newContent = newContent.replace(/from-brand-primary-[0-9]+/g, 'from-brand-primary');
            newContent = newContent.replace(/to-brand-primary-[0-9]+/g, 'to-brand-primary');
            newContent = newContent.replace(/via-brand-primary-[0-9]+/g, 'via-brand-primary');
            newContent = newContent.replace(/from-brand-secondary-[0-9]+/g, 'from-brand-secondary');
            newContent = newContent.replace(/to-brand-secondary-[0-9]+/g, 'to-brand-secondary');

            if (content !== newContent) {
                fs.writeFileSync(fullPath, newContent, 'utf8');
                console.log(`Updated ${fullPath}`);
            }
        }
    }
}

processDirectory(directory);
console.log('Update complete.');
